subtests/docker_cli/liverestore/liverestore.py

```diff
@@ -0,0 +1,200 @@
+See rhbz1424709#c40 and rhbz1460266.
+
+"""
+
+import time
+from autotest.client import utils
+import dockertest.docker_daemon as docker_daemon
+from dockertest import subtest
+from dockertest.containers import DockerContainers
+from dockertest.dockercmd import AsyncDockerCmd, DockerCmd
+from dockertest.images import DockerImage
+from dockertest.output import DockerInfo
+from dockertest.output.validate import mustpass
+from dockertest.xceptions import DockerTestNAError
+
+
+class liverestore(subtest.Subtest):
+
+    def initialize(self):
+        super(liverestore, self).initialize()
+
+        # Skip test if live-restore is not enabled
+        try:
+            lr_enabled = DockerInfo().get('Live Restore Enabled')
+            self.failif_ne(lr_enabled, 'true',
+                           "Live Restore Enabled field from 'docker info'",
+                           DockerTestNAError)
+        except KeyError:
+            # docker 1.12 doesn't include this in 'docker info'. Try looking
+            # at command line. This won't work with containerized docker
+            # because it uses /etc/docker/*.json
+            docker_cmdline = docker_daemon.cmdline()
+            self.failif_not_in("--live-restore", docker_cmdline,
+                               "docker daemon command-line options",
+                               DockerTestNAError)
+
+        # rhbz1424709#c40 : live-restore fails unless KillMode=process
+        killmode = docker_daemon.systemd_show('KillMode')
+        if killmode != 'process':
+            self.logwarning("systemd KillMode is '%s' (expected 'process')."
+                            " This test will probably fail." % killmode)
+
+        self.stuff['dc'] = DockerContainers(self)
+
+    def run_once(self):
+        super(liverestore, self).run_once()
+
+        self._start_container()
+
+        # Verify that the container is running and that it's a child
+        # of the main docker daemon
+        self._verify_that_date_changes()
+        self.failif(not self._container_is_child_of_docker_daemon(),
+                    "Newly-created container is not a descendant of dockerd")
+
+        # Restart docker daemon. It should be nearly instantaneous; if it
+        # takes ~90 seconds it could be another symptom of rhbz1424709.
+        # The warning message might help someone diagnose a later failure.
+        self.stuff['dockerd_pid_orig'] = docker_daemon.pid()
+        self.stuff['container_pid_orig'] = self._container_pid()
+        t0 = time.time()
+        docker_daemon.restart()
+        t1 = time.time()
+        if t1 - t0 > 30:
+            self.logwarning("docker restart took %d seconds", t1 - t0)
+
+    def _start_container(self):
+        """
+        Start a container. We only need it briefly, until we (the test,
+        running in host space) can cd into its root filesystem. Container
+        will spin idly until a semaphore file is removed.
```

Comment:
Nowhere in this function is anything acting on a lock ifle, should this part of the docstring be moved elsewhere?